connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -487,16 +487,27 @@ export async function syncDeltaForRootNodesInDrive({
   });
   const uniqueChangedItems = removeAllButLastOccurences(results);
 
-  const microsoftNodes = await concurrentExecutor(
-    rootNodeIds,
-    async (rootNodeId) =>
-      getItem(client, typeAndPathFromInternalId(rootNodeId).itemAPIPath),
-    { concurrency: 5 }
+  const sortedChangedItems: DriveItem[] = [];
+  const containWholeDrive = rootNodeIds.some(
```

Comment:
nit: what about keeping the code exactly as before, but passing the drive root folder id as `rootId` when we chose the full drive?
optional of course, if you think it's worth it

Reply:
it avoid calling getItem , and make a single call to sortForIncrementalUpdate even if we have sub folders selected (which actually happen)